src/plugin_host.jl

```diff
@@ -0,0 +1,174 @@
+start_host() = try wait(nvim_child(HostHandler()).reader) end
+
+immutable HostHandler
+    loaded_plugins::Set{String} # plugin file paths
+    proc_callbacks::Dict{String, Function}
+end
+
+function HostHandler()
+    HostHandler(Set{String}(), Dict{String, Function}())
+end
+
+# names/methods for invoked cmds/fns have the form:
+#   <file path>:(command|function|autocmd):<procedure name>
+function on_notify(h::HostHandler, c, name::String, args::Vector{Any})
+    proc = require_plug(h, name)
+
+    if proc == nothing
+        throw(ErrorException(name * " not defined"))
+    end
+
+    @async(try
+        proc(c, args...)
+    catch
+        throw(MethodError(name, tuple(NvimClient, args...)))
+        throw(err)
```

Comment:
I think it is ` ... catch err; showerror(STDERR, err, catch_backtrace());` let's also print "while handling notification" and the name,args for contexts.